cmd/openshift-install/gather.go

```diff
@@ -95,30 +96,30 @@ func runGatherBootstrapCmd(directory string) error {
 	if err != nil {
 		if err2, ok := err.(errUnSupportedGatherPlatform); ok {
 			logrus.Error(err2)
-			return unSupportedPlatformGather()
+			return unSupportedPlatformGather(directory)
 		}
 		return errors.Wrapf(err, "failed to get bootstrap and control plane host addresses from %q", tfStateFilePath)
 	}
 
-	logGatherBootstrap(bootstrap, port, masters)
+	logGatherBootstrap(bootstrap, port, masters, directory)
 	return nil
 }
 
-func logGatherBootstrap(bootstrap string, port int, masters []string) {
-	var (
-		sshport string
-		scpport string
-	)
-	if port != 22 {
-		sshport = fmt.Sprintf(" -p %d", port)
-		scpport = fmt.Sprintf(" -P %d", port)
+func logGatherBootstrap(bootstrap string, port int, masters []string, directory string) {
+	logrus.Info("Pulling logs from bootstrap for debugging")
+
+	client, err := ssh.NewClient("core", fmt.Sprintf("%s:%d", bootstrap, port), nil)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+	if err := ssh.Run(client, fmt.Sprintf("/usr/local/bin/installer-gather.sh %s", strings.Join(masters, " "))); err != nil {
+		logrus.Fatal(err)
 	}
-	if s, ok := os.LookupEnv("SSH_AUTH_SOCK"); !ok || s == "" {
-		logrus.Info("Make sure ssh-agent is running, env SSH_AUTH_SOCK is set to the ssh-agent's UNIX socket and your private key is added to the agent.")
+	lFile := filepath.Join(directory, "log-bundle.tar.gz")
+	if err := ssh.PullFileTo(client, "/home/core/log-bundle.tar.gz", lFile); err != nil {
```

Comment:
Fixed filename is going to be racy in the face of multiple, simultaneous gathers.  Probably not a high risk, but do we want to address that here anyway by adding an optional environment variable or some such to set a custom filename, seeded with some random characters in Go here?

Reply:
done with https://github.com/openshift/installer/pull/1822/commits/cad7f02a8bdfc268ee54ab4f4e25dec1a99640c3
```console
./bin/openshift-install --dir dev gather bootstrap --log-level debug --bootstrap 192.168.126.10 --master 192.168.126.11
DEBUG OpenShift Installer unreleased-master-1103-g55e406d64e8c57a53c83d677f6ac936dda6de19f
DEBUG Built from commit 55e406d64e8c57a53c83d677f6ac936dda6de19f
INFO Pulling debug logs from the bootstrap machine
DEBUG Gathering bootstrap journals ...
DEBUG Gathering bootstrap containers ...
DEBUG Gathering rendered assets...
DEBUG Gathering cluster resources ...
DEBUG Waiting for logs ...
.....
DEBUG Gather remote logs
DEBUG Collecting info from 192.168.126.11
DEBUG lost connection
 EBUG ssh: connect to host 192.168.126.11 port 22: No route to host
DEBUG Log bundle written to ~/log-bundle.tar.gz
INFO Bootstrap gather logs captured here "dev/log-bundle-20190606143715.tar.gz"
```
Does this look better..?

Reply:
> ```
> INFO Bootstrap gather logs captured here "dev/log-bundle-20190606143715.tar.gz"
> ```

You've fixed installer-side collisions, which is good.  But I was worried about bootstrap-side collisions:

1. Alice runs `gather bootstrap`, her command writes a tarball on the bootstrap machine and starts downloading it.
2. Bob runs `gather bootstrap`, his command starts writing to the same tarball on the bootstrap machine.
3. Alice's download croaks on the truncated file.

But maybe Alice's download already has an open file descriptor, so Bob's clobber doesn't break the download?  I guess it depends on the specifics of how `installer-gather.sh`'s `> ~/log-bundle.tar.gz` works.  But I'd be happier if that was:

```sh
TAR_FILE="${TAR_FILE:-${HOME}/log-bundle.tar.gz}"
tar cz -C /tmp/artifacts . >"${TAR_FILE}"
echo "Log bundle written to ${TAR_FILE}"
```

with a SSH command like:

```go
bootstrapFile := fmt.Sprintf("/tmp/log-bundle-%s.tar.gz", time.Now().Format("20060102150405"))
if err := ssh.Run(client, fmt.Sprintf("TAR_FILE=%s /usr/local/bin/installer-gather.sh %s", bootstrapFile, strings.Join(masters, " "))); err != nil {
```

Reply:
```
tar cz -C /tmp/artifacts . >"${TAR_FILE}"
```
Hmm, that's vulnerable to simultaneous gathers too.  But it's a smaller cross-section.  I'm fine punting all of these bootstrap-collision fixes to follow-up work if you don't want to pick them up here.

Reply:
This is not something this PR is about. collision on the bootstrap host script should be handled in separate PR.

Reply:
Spun off into #1836.